### DISCUSSION TITLE
Possible to pull github actions images?

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#26605] [Actions]
<p>All,</p>
<p>This might be a FAQ and my search-fu is failing me, but I couldn’t find it. Namely, sometimes I need to try and debug what’s happening in my GitHub Actions when they fail. Unlike CircleCI, I can’t ssh into the run, so my thought was “Can I pull the image locally and issue the action commands?”</p>
<p>That is, is it possible to <code>docker pull github-actions/ubuntu-20.04</code> or whatever my yaml file is asking for and get the image that GitHub Actions uses?</p>
<p>Thanks,<br>
Matt</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There is no official Docker image for that, because the GitHub runners are VMs. You should be able to find more about them here: <a href="https://github.com/actions/virtual-environments" class="inline-onebox">GitHub - actions/virtual-environments: GitHub Actions virtual environments</a></p>
<p>There is a (non GitHub!) tool to run Actions locally using Docker, and there are some Docker images mentioned in its README. I haven’t used them, so I can’t say exactly how close to the VMs they are, but you can take a look here: <a href="https://github.com/nektos/act#runners" class="inline-onebox">GitHub - nektos/act: Run your GitHub Actions locally 🚀</a></p>

**Comment by mathomp4** [comment]
<p>Ohhh. I feel a bit dumb. I thought maybe I just didn’t know Docker speak well and it was just a GitHub-y way of talking about containers, etc. Explains why I couldn’t find anyone saying "just do <code>docker pull ghcp.io/...</code></p>
<p>Maybe I should go find a “Please let us ssh in” issue and vote on that! :slight_smile:</p>

**Comment by Simran-B** [comment]
<p>You may also give this action a try to ssh into the virtual environment: <a href="https://github.com/marketplace/actions/debugging-with-tmate" class="inline-onebox" rel="noopener nofollow ugc">Debugging with tmate · Actions · GitHub Marketplace · GitHub</a></p>
